internal/dag/httpproxy_processor.go

```diff
@@ -1026,6 +1031,7 @@ func (p *HTTPProxyProcessor) computeRoutes(
 				SlowStartConfig:               slowStart,
 				MaxRequestsPerConnection:      p.MaxRequestsPerConnection,
 				PerConnectionBufferLimitBytes: p.PerConnectionBufferLimitBytes,
+				UpstreamTLS:                   determineUpstreamTLS(p.UpstreamTLS),
```

Comment:
I _think_ you can use the same casting you're using in some of the other processors and drop the helper function entirely, though functionally it's just fine as-is.